db/migrate/20210202052337_migrate_variant_unit_values.rb

```diff
@@ -1,11 +1,13 @@
 class MigrateVariantUnitValues < ActiveRecord::Migration
   def up
-    Spree::Variant.all.select { |v|
-      v.unit_value.nil? && v.product&.variant_unit == "items"
-    }.each do |variant|
+    Spree::Variant.includes(:product).where(
+      spree_products: { variant_unit: "items" },
+      spree_variants: { unit_value: nil }
+    ).each do |variant|
```

Comment:
Sorry I didn't mention it but we'll also need to use `#find_each` if can't avoid instantiating the variants to run callbacks.